docs/data/material/customization/containerQueries/containerQueries.md

```diff
@@ -0,0 +1,91 @@
+# Container Queries
+
+<p class="description">API that enables the use of CSS container queries based on theme breakpoints.</p>
+
+A utility function based on the [theme breakpoints](/material-ui/customization/breakpoints/) for creating [CSS container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries).
+
+## Usage
+
+To create a container query, use `theme.containerQueries` with any method available in the [theme.breakpoints](/material-ui/customization/breakpoints/#api).
+The value can be a unitless (pixel), a string, or a breakpoint key, for example:
+
+```js
+theme.containerQueries.up('sm'); // => '@container (min-width: 600px)'
+```
+
+:::info
+One of the ancestors must have the CSS container type specified.
+:::
+
+{{"demo": "BasicContainerQueries.js"}}
+
+### Named containment contexts
+
+To refer to a [containment context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_containment/Container_queries#naming_containment_contexts), call the `containerQueries` method with the name, then you will be able to access all the breakpoint methods like the example above:
+
+```js
+theme.containerQueries('sidebar').up('500px'); // => '@container sidebar (min-width: 500px)'
+```
+
+## Shorthand syntax
+
+In sx prop, a format of `@<size>` or `@<size>/<name>` can be used to apply container queries without refering to the theme.
```

Comment:
Unrelated: I can't wait for a future where we can use the same syntax in `styled` and `sx`.

Reply:
It sounds possible to me for v7 (only with Pigment CSS).